api/envoy/config/filter/network/redis_proxy/v2/redis_proxy.proto

```diff
@@ -82,6 +83,21 @@ message RedisProxy {
     // If `max_buffer_size_before_flush` is set, but `buffer_flush_timeout` is not, the latter
     // defaults to 3ms.
     google.protobuf.Duration buffer_flush_timeout = 5 [(gogoproto.stdduration) = true];
+
+    // If `enable_redirection` is true, upstream connections to hosts that are not part of cluster
+    // configuration may be created. If these hosts are later discovered to be part of the cluster
+    // (e.g., redis cluster) or a host is removed from the cluster (whether `enable_redirection` is
+    // true or not), then the associated connections are drained and closed.
+    // `upstream_drain_poll_interval` controls how often the draining connections are checked to see
+    // if they are inactive before being closed; this interval defaults to 100ms.
+    google.protobuf.Duration upstream_drain_poll_interval = 6 [(gogoproto.stdduration) = true];
```

Comment:
Q: Why do we need a timer here for polling? Can't we just get a callback when the connection is empty?

Reply:
we could, but I didn't want to add a queue check into the server response data path for all response processing -- when draining should be rare or even never happen....

Reply:
Fair enough, but then does this really need to be configurable? Can we just skip the configuration and pick an interval in code?

Reply:
no, it doesn't need to be configurable. I was just anticipating the desire for a control....which has happened before :).

Reply:
I'll make it a static value of a second; there is no need to be aggressive in cleaning up the list.

Reply:
+1, if you are concerned, you can allow the value to be loaded from runtime. IMO that is good enough.